### PR TITLE
Use print() function in both Python 2 and Python 3

### PR DESCRIPTION
**Pull request by cclauss**
__print()__ is a function in Python 3.

### CODE REVIEW

File: satpy/readers/nc_olci.py [hunk 1]
```diff
@@ -33,6 +33,7 @@
 from satpy.readers.file_handlers import BaseFileHandler
 from satpy.utils import angle2xyz, xyz2angle
 from satpy import CHUNK_SIZE
+from functools import reduce
```
Comment (djhoese) [inline satpy/readers/nc_olci.py]: This has been fixed in #515 but has not been merged yet. Please remove it from this PR and we'll include it from that PR.

File: satpy/tests/reader_tests/test_nc_olci.py [hunk 2]
```diff
@@ -23,6 +23,7 @@
 """Module for testing the satpy.readers.nc_olci module.
 """
 import sys
+from functools import reduce
```
[the remark this answers is not in the document]
Comment (djhoese) [reply satpy/tests/reader_tests/test_nc_olci.py]: As mentioned by stickler, please remove this import.

File: satpy/dataset.py [hunk 3]
```diff
@@ -201,9 +201,9 @@ def wavelength_match(a, b):
             a (tuple or scalar): (min wl, nominal wl, max wl) or scalar wl
             b (tuple or scalar): (min wl, nominal wl, max wl) or scalar wl
         """
-        if type(a) == (type(b) or
+        if isinstance(a, (type(b) or
                        isinstance(a, numbers.Number) and
-                       isinstance(b, numbers.Number)):
+                       isinstance(b, numbers.Number))):
```
[the remark this answers is not in the document]
Comment (djhoese) [reply satpy/dataset.py]: I don't think this is the same behavior.

**Comment by djhoese** [comment]
Why close this?

**Comment by cclauss** [comment]
Closed in favor of #515 and #517

**Comment by djhoese** [comment]
But those PRs don't fix the prints in the scripts that you fixed.

**Comment by cclauss** [comment]
Oh, OK.  Please add those mods to #515 because most reviewers will assume that all==all.

**Comment by djhoese** [comment]
As mentioned in the PR these checks were done on `flake8 satpy`. The point being the actual package code and not necessarily the utility modules, but you make a good point. If you still have these print statement fixes I'd be happy to merge them if you re-make the PR.

**Comment by cclauss** [comment]
> I ran flake8 satpy from the root of the repository...

**Pull request closed without merging**